### ISSUE TITLE
`tflint-ignore: all` not ignoring error

### ISSUE THREAD

**Issue by Carles-Figuerola** [terraform-linters/tflint#942]
Example code:

```hcl
data "local_file" "main" {
    filename = "main.tf"
}

resource "null_resource" "tflint-test" {

  provisioner "local-exec" {
    command     = "echo"
    environment = {
      testenv = data.local_file.main.content
    }
  }

  provisioner "local-exec" {
    when        = destroy
    command     = "echo"
    environment = {
      # tflint-ignore: all
      testenv = data.local_file.main.content
    }
  }
}
```

Error shown:
```
tflint .
Failed to load configurations. 1 error(s) occurred:

Error: Invalid reference from destroy provisioner

  on main.tf line 17, in resource "null_resource" "tflint-test":
  17:     environment = {
  18:       # tflint-ignore: all
  19:       testenv = data.local_file.main.content
  20:     }

Destroy-time provisioners and their connection configurations may only reference attributes of the related resource, via 'self', 'count.index', or 'each.key'.

References to other resources during the destroy phase can cause dependency cycles and interact poorly with create_before_destroy.
```

## Version
TFLint v0.20.2

**Comment by bendrucker** [comment]
`tflint-ignore` doesn't reject errors, just rule violations. Are you sure Terraform will even allow this? That's where this error is coming from, as opposed to TFLint itself.

**Comment by Carles-Figuerola** [comment]
Thanks for the quick response, I see now where this is coming from now. Running this as a plan in terraform `0.12.x` shows this:

```
Warning: External references from destroy provisioners are deprecated

  on main.tf line 17, in resource "null_resource" "tflint-test":
  17:     environment = {
  18:       # tflint-ignore: all
  19:       testenv = data.local_file.main.content
  20:     }

Destroy-time provisioners and their connection configurations may only
reference attributes of the related resource, via 'self', 'count.index', or
'each.key'.

References to other resources during the destroy phase can cause dependency
cycles and interact poorly with create_before_destroy.
```

whereas terraform `0.13.x` show this:
```
Error: Invalid reference from destroy provisioner

  on main.tf line 17, in resource "null_resource" "tflint-test":
  17:     environment = {
  18:       # tflint-ignore: all
  19:       testenv = data.local_file.main.content
  20:     }

Destroy-time provisioners and their connection configurations may only
reference attributes of the related resource, via 'self', 'count.index', or
'each.key'.
```

We haven't upgraded to terraform `0.13.x` yet as we have a big codebase, and as tflint doesn't show what terraform version it's based on, we are getting this error.

Is there any way to change this behaviour on run rather than using an older version? (just so we can upgrade one module at a time and still use the same tflint job)

**Comment by bendrucker** [comment]
> Is there any way to change this behaviour on run rather than using an older version? (just so we can upgrade one module at a time and still use the same tflint job)

Not possible, sorry. TFLint relies on Terraform's own libraries for loading/parsing configuration and thus is tied to a given Terraform version with the accompanying restrictions.
